### PR TITLE
Dto objects, business services

### PR DESCRIPTION
**Pull request by ArtyomKolosov2**

### CODE REVIEW

**Review by dbogdan2000 — APPROVED** [review]

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IDirectionService.cs [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
     {
-        Task<DirectionModel> Create(DirectionModel direction);
-        Task<DirectionModel> GetById(int id);
-        Task<IEnumerable<DirectionModel>> GetAll();
-
-        Task<DirectionModel> Update(DirectionModel direction);
-
-        Task Delete(int id);
+        public bool Equal(DirectionModel directionModel1, DirectionModel directionModel2);
+        public bool NotEqual(DirectionModel directionModel1, DirectionModel directionModel2);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IDirectionService.cs:12]: Hmm... Why do you need it?

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserProfileService.cs [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
-
-        Task<UserProfileModel> GetById(int id);
-
-        Task<IEnumerable<UserProfileModel>> GetAll();
-
-        Task<UserProfileModel> Update(UserProfileModel userProfile);
-
-        Task Delete(int id);
+        public bool Equal(UserProfileModel userModel1, UserProfileModel userModel2);
+        public bool NotEqual(UserProfileModel userModel1, UserProfileModel userModel2);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserProfileService.cs:12]: Again don't see the reason

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/ReadOnlyService.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public abstract class ReadOnlyService<TModel, TEntity, TRepository> : IReadOnlyService<TModel>
+        where TModel : class
+        where TEntity : class
+        where TRepository : IReadOnlyRepository<TEntity>
+    {
+        protected readonly IMapper _mapper;
+        protected readonly TRepository _repository;
+        protected readonly IUnitOfWork _unitOfWork;
+
+        protected ReadOnlyService(TRepository repository, IUnitOfWork unitOfWork, IMapper mapper)
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Services/ReadOnlyService.cs:22]: Choose pls what will you use repo or UoW. Don't use both ways

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/Service.cs [hunk 4]
```diff
@@ -43,6 +91,8 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        
+
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Services/Service.cs:95]: Empty spaces

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi!
I did some comments. Let me know if any questions.

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/ReadOnlyService.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (_repository is Repository<TEntity> abstractRepository)
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Services/ReadOnlyService.cs:59]: As I told need to add disposable behaviour to IRepository and IReadOnlyRepository

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/Service.cs [hunk 6]
```diff
@@ -1,25 +1,73 @@
...
             _unitOfWork = unitOfWork;
             _mapper = mapper;
         }
 
+        public async Task<TModel> Create(TModel model)
+        {
+            var entity = _mapper.Map<TEntity>(model);
+
+            var createdEntity = await _repository.Create(entity);
+            await _unitOfWork.SaveChanges();
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Services/Service.cs:32]: You don't need UoW for saving action. You can add it in base repository class

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Some comments for you, dude)

**Review by PushMeNow — APPROVED** [review]
Approved

**Pull request merged**